docs/modules/ROOT/pages/api/account.adoc

```diff
@@ -0,0 +1,213 @@
+* xref:#Account-validate_transaction[`++validate_transaction(self)++`]
+* xref:#Account-_set_public_key[`++_set_public_key(self, new_public_key)++`]
+* xref:#Account-_is_valid_signature[`++_is_valid_signature(self, hash, signature)++`]
+--
+
+[.contract-index]
+.Events
+--
+* xref:#Account-OwnerAdded[`++OwnerAdded(new_owner_guid)++`]
+* xref:#Account-OwnerRemoved[`++OwnerRemoved(removed_owner_guid)++`]
+--
+
+[.contract-item]
+[[Account-__validate_deploy__]]
+==== `[.contract-item-name]#++__validate_deploy__++#++(self: @ContractState, class_hash: felt252, contract_address_salt: felt252, _public_key: felt252) → felt252++` [.item-kind]#external#
+
+Validates a DeployAccount transaction.
+
+Returns the short string `'VALID'` if valid, or any other value otherwise.
+
+[.contract-item]
+[[Account-__execute__]]
+==== `[.contract-item-name]#++__execute__++#++(ref self: ContractState, calls: Array<Call>) → Array<Span<felt252>>++` [.item-kind]#external#
+
+See xref:ISRC6-\\__execute__[ISRC6::\\__execute__].
+
+[.contract-item]
+[[Account-__validate__]]
+==== `[.contract-item-name]#++__validate__++#++(self: @ContractState, calls: Array<Call>) → felt252++` [.item-kind]#external#
+
+See xref:ISRC6-\\__validate__[ISRC6::\\__validate__].
+
+[.contract-item]
+[[Account-is_valid_signature]]
+==== `[.contract-item-name]#++is_valid_signature++#++(self: @ContractState, hash: felt252, signature: Array<felt252>) → felt252++` [.item-kind]#external#
+
+See xref:ISRC6-is_valid_signature[ISRC6::is_valid_signature].
+
+[.contract-item]
+[[Account-supports_interface]]
+==== `[.contract-item-name]#++supports_interface++#++(self: @ContractState, interface_id: felt252) → bool++` [.item-kind]#external#
+
+Returns whether a contract implements a given interface or not.
+
+[.contract-item]
+[[Account-__validate_declare__]]
+==== `[.contract-item-name]#++__validate_declare__++#++(self: @ContractState, class_hash: felt252) → felt252++` [.item-kind]#external#
+
+Validates a Declare transaction.
+
+Returns the short string `'VALID'` if valid, or any other value otherwise.
+
+[.contract-item]
+[[Account-set_public_key]]
+==== `[.contract-item-name]#++set_public_key++#++(ref self: ContractState, new_public_key: felt252)++` [.item-kind]#external#
+
+Sets a new public key for the account. Can be called only by self through the `\\__execute__` mechanism.
+
+Emits an {OwnerRemoved}, and an {OwnerAdded} event.
+
+[.contract-item]
+[[Account-get_public_key]]
+==== `[.contract-item-name]#++get_public_key++#++(self: @ContractState)++ → felt252` [.item-kind]#external#
+
+Returns the current public key of the account.
+
+[.contract-item]
+[[Account-initializer]]
+==== `[.contract-item-name]#++initializer++#++(ref self: ContractState, _public_key: felt252)++` [.item-kind]#internal#
+
+Initializes the account with the given public key, and registers the ISRC6 interface ID.
+
+Emits an {OwnerAdded} event.
+
+[.contract-item]
+[[Account-validate_transaction]]
```

Comment:
i wonder if there's anything we can do to visually separate externals from internals from events, etc

Reply:
i like it! i think we can further distinguish between externals and internals, no?

Reply:
```suggestion
Returns the short string `'VALID'` if valid, otherwise it reverts.
```

Reply:
I like it, but unsure about "Internalimpl", it looks just like another impl. How can we properly communicate which functions are external and which internals visually?

Looking forward, is it possible that we offer a component (not a contract/preset) with both externals and internals? or components will always be externals?

Reply:
From what I recall from previous conversations. Every method in a component will be internal (or private, etc..). External methods needs to be exposed in the contract using the component, probably by using the component API.

Important to note that external in this context means callable as target for a transaction, not usable by others module in the cairo code.

Reply:
> How can we properly communicate which functions are external and which internals visually?

Maybe we can split the `Functions` in `External Functions` and `Internal Functions`? Not all the internal functions need to be in the InternalImpl (they will most of the time though)

Reply:
I updated the Account API to reflect this proposal, lmk wdyt